File: week01/Theme02/Theme02_Code02.cpp
// Theme02_Code02.cpp

#include <iostream>
#include <queue>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	queue<int> q;

	q.push(0);
	q.push(1);
	q.push(2);

	cout << q.front() << " " << q.back() << endl; // 0 2
	
	q.push(3);
	q.push(4);
	q.pop();
	q.pop();

	cout << q.front() << " " << q.back() << endl; // 2 4
}

File: week01/Theme02/Theme02_Code02.py
# Theme01_Code02.py

import queue

q = queue.Queue()
    
q.put(0)
q.put(1)
q.put(2)

print(q.queue[0], q.queue[-1]) # 0 2

q.put(3)
q.put(4)
q.get()
q.get()

print(q.queue[0], q.queue[-1]) # 2 4